rpc-rs/src/otel.rs

```diff
@@ -18,29 +18,41 @@ lazy_static::lazy_static! {
 #[derive(Debug)]
 pub struct OtelHeaderExtractor<'a> {
     inner: &'a HeaderMap,
+    keys: Vec<String>,
```

Comment:
HeaderName doesn't have `as_str`, only `as_ref() -> &[u8]` (even though it's stored as String internally).
To implement Extractor trait's `keys(&self) -> Vec<&str>`, we can't use String.from_utf8 because the return value wouldn't be owned by self. So we have to maintain a separate internal String copy of all keys